packages/test-helper/src/testHelper.ts

```diff
@@ -0,0 +1,428 @@
+import { HttpClientInterface } from '@apimatic/core-interfaces';
+import { Readable } from 'stream';
+
+/**
+ * Get streaming data from a given URL.
+ * @param client Instance of HttpClient to be used.
+ * @param url URL from which to create the readable stream.
+ * @returns Stream of data fetched from the URL.
+ * @throws Error if unable to retrieve data from the URL.
+ */
+export async function getStreamData(client: HttpClientInterface, url: string) {
+  const res = await client({
+    method: 'GET',
+    url,
+    responseType: 'stream',
+  });
+  if (res.statusCode !== 200 || typeof res.body === 'string') {
+    throw new Error(`Unable to retrieve streaming data from ${url}`);
+  }
+  return res.body as any;
+}
+
+/**
+ * Check if input matches the contents of a file by comparing stream data or Blob data.
+ * @param expected Expected data (NodeJS ReadableStream or Blob).
+ * @param actual Input data to compare against file contents (NodeJS ReadableStream or Blob).
+ * @returns Promise resolving to true if input matches file contents, otherwise false.
+ */
+export async function areStreamsMatching(
+  expected: NodeJS.ReadableStream | Blob,
+  actual: NodeJS.ReadableStream | Blob | undefined
+): Promise<boolean> {
+  if (typeof actual === 'undefined') {
+    return false;
+  }
+  try {
+    const expectedBuffer =
+      expected instanceof Readable
+        ? await streamToBuffer(expected)
+        : await blobToBuffer(expected as Blob);
+
+    const actualBuffer =
+      actual instanceof Readable
+        ? await streamToBuffer(actual)
+        : await blobToBuffer(actual as Blob);
+
+    return Buffer.compare(actualBuffer, expectedBuffer) === 0;
+  } catch (error) {
+    return false;
+  }
+}
+
+/**
+ * Compare actual headers with expected headers, ignoring case sensitivity.
+ * @param actualHeaders Actual headers received from the request.
+ * @param expectedHeaders Expected headers with values to match against actual headers.
+ */
+export function expectHeadersToMatch(
+  actualHeaders: Record<string, string>,
+  expectedHeaders: Record<string, Array<string | boolean>>
+) {
+  const lowerCasedHeaders = Object.keys(actualHeaders).reduce((acc, key) => {
+    acc[key.toLowerCase()] = actualHeaders[key];
+    return acc;
+  }, {} as Record<string, string>);
+
+  Object.entries(expectedHeaders).forEach(([expectedKey, expectedValue]) => {
+    const lowerCasedKey = expectedKey.toLowerCase();
+    expect(lowerCasedHeaders).toHaveProperty(lowerCasedKey);
+    if (expectedValue[1]) {
+      expect(lowerCasedHeaders[lowerCasedKey]).toBe(expectedValue[0]);
+    }
+  });
+}
+
+/**
+ * Recursively check whether the left object or array is a proper subset of the right object or array.
+ * @param left Left object or array.
+ * @param right Right object or array.
+ * @param options Options for subset comparison.
+ * @returns Boolean indicating if left is a proper subset of right.
+ */
+export function isProperSubsetOf(
+  left: any,
```

Comment:
@MaryamAdnan3 CodeClimate is reporting Cognitive Complexity and other complex code issues, its also reporting that the code coverage is below par in some of the following helper functions